src/core/io/writer/h5md/h5md_core.hpp

```diff
@@ -39,180 +38,187 @@
-        /*
-         * @brief Method to write the simulation script to the dataset.
-         */
-        void WriteScript(std::string const &filename);
-
-        /**
-         * @brief Creates a new H5MD file.
-         * @param filename The filename
-         */
-        void create_new_file(const std::string& filename);
-
-        /**
-         * @brief Loads an existing H5MD file.
-         * @param filename The filename
-         */
-        void load_file(const std::string& filename);
-
-        /**
-         * @brief Initializes the necessary data to create the datsets and attributes.
-         */
-        void init_filestructure();
-
-        /**
-         * @brief Creates the necessary HDF5 datasets.
-         * @param only_load Set this to true if you want to append to an existing file.
-         */
-        void create_datasets(bool only_load);
-        
-        /**
-         * @brief Links the time and step datasets to of all properties to the time and step dataset of the id property. All properties are written at the same time.
-         */
-	    void create_links_for_time_and_step_datasets();
-
-        /**
-         * Member variables.
-         */
-    	int m_max_n_part = 0;
-        std::string m_filename;
-        std::string m_scriptname;
-        int m_what;
-        bool m_write_ordered;
-        std::string m_backup_filename;
-        boost::filesystem::path m_absolute_script_path = "NULL";
-        h5xx::file m_h5md_file;
-
-        struct DatasetDescriptor  {
-            std::string path;
-            hsize_t dim;
-            h5xx::datatype type;
-        };
-        std::vector<std::string> group_names;
-        std::vector<DatasetDescriptor> dataset_descriptors;
-        std::unordered_map<std::string, h5xx::dataset> datasets;
+class File {
+public:
+  /**
+   * Constructor/destructor without arguments (due to script_interface).
+   * @brief Constructor of the "File" class.
+  */
+  File();
+  ~File();
+  /**
+   * @brief Initialize the File object.
+   */
+  void InitFile();
+  /**
+   * @brief Method to perform the renaming of the temporary file from
+   * "filename" + ".bak" to "filename".
+   */
+  void Close();
+
+  enum WriteData {
+    W_POS = 1 << 0,
+    W_V = 1 << 1,
+    W_F = 1 << 2,
+    W_TYPE = 1 << 3,
+    W_MASS = 1 << 4,
+    W_CHARGE = 1 << 5
+  };
+  /**
+   * @brief General method to write to the datasets which calls more specific
+   * write methods.
+   * @param Boolean values for position, velocity, force and mass.
+   */
+  void Write(int write_dat);
+
+  /**
+   * @brief Method to write the energy contributions to the H5MD file.
+   * @param Boolean values for total, kinetic.
+   * \todo Implement this method.
+   */
+  void WriteEnergy(bool total = true, bool kinetic = true);
+  std::string &filename() { return m_filename; };
```

Comment:
this is due to the scriptinterface access, I just adopted this structure from e.g. shapes/Wall.hpp.